### DISCUSSION TITLE
Grouped GitLens views in Source Control in GitLens 16

### DISCUSSION THREAD

**Discussion by justinrobots** [gitkraken/vscode-gitlens#3762] [Announcements]
In GitLens 16, we created one consolidated `GitLens` view in Source Control that contains `Commits`, `Branches`, `Remotes`, `Stashes`, `Tags`, `Worktrees`, and `Contributors`. We did this to reduce noise in Source Control and allow more customization.

<img width="1188" alt="ungrouped-vs-grouped" src="https://github.com/user-attachments/assets/e835b872-2bd0-440f-b97e-74dba31ab549">

This one-time change was necessary to create a consolidated view, but there are a few simple ways to put views back where they were if you can't find them or prefer them in separate locations:
1. In the new `GitLens` view in `Source Control` you will see a one-time message that asks if you want to continue with all of the views grouped, or restore them. Choose "Restore views to previous locations" to keep views where they were.
<p align="center"><img width="455" alt="restore-previous" src="https://github.com/user-attachments/assets/2c01a69c-b93e-4373-9dcb-67486292b307"></p>

2. At any time, you can select `Detach All Views` in the `...` menu of the `GitLens` view and all views will be revealed where you last had placed them.
<p align="center"><img width="681" alt="detach-all" src="https://github.com/user-attachments/assets/8ae02abf-2f23-405c-8495-029b331979f0"></p>

You can continue to customize your GitLens layout by **detaching** views from, or **regrouping** views to the `GitLens` view. Individual views can be `regrouped` by clicking the `x` in the view header.
<p align="center"><img width="490" alt="regroup-single" src="https://github.com/user-attachments/assets/c4093303-e6ff-462b-a1e6-64d7b525b0e7"></p>

**Comment by hengistchan** [comment]
breaking UI changes

**Comment by coconewty** [comment]
this should have been released as an opt-in feature, not forced on users. I just wasted a chunk of time trying to work out where my tabs just disappeared to

**Reply by justinrobots** [reply]
Hi @coconewty, thanks for sharing your feedback, and apologies for the disruption. When we introduced this feature, our goal was to balance discoverability with minimal interruption. Given our limitations as an extension, we opted for a "welcome" message in the Source Control view with a quick opt-out option, rather than a pop-up, to avoid disrupting workflows.

However, due to recent changes in VS Code’s view controls, the message wasn’t displayed as intended for some users. We’ve since implemented a workaround to make this more clear for users who haven’t yet upgraded. This kind of feedback helps us refine how we communicate changes to GitLens, so thanks for reaching out.

**Reply by sebastian-correa** [reply]
While I also wasted time looking around for tabs [[ref](https://github.com/gitkraken/vscode-gitlens/issues/3766#issuecomment-2483329942)] and think there would've been better ways to handle this (though I really don't know what can and can't be done as a VSCode extension), I want to commend the easy-to-find issue and documentation for reverting the UI, the openness to feedback, and the actually good feature (you can now hide unwanted stuff under a single tab), chaotic as it might've been for some users to adopt.

**Comment by MorningPants** [comment]
![image](https://github.com/user-attachments/assets/1d9aa6e0-2807-4cf6-80c5-b44d7af42453)
Was the update supposed to remove the branches tab?

**Reply by MorningPants** [reply]
And how do I get it back?

**Reply by justinrobots** [reply]
@MorningPants it looks like your `Source Control` is hidden in the sidebar, but this shouldn't have been caused by GitLens. You can bring it back by right-clicking on one of the icons in the sidebar and checking `Source Control`.

<img width="452" alt="image" src="https://github.com/user-attachments/assets/12a3c7f3-5ee5-42de-a17b-a8a0ef562c0f">

**Reply by MorningPants** [reply]
Thanks, I was able to restore it.  Somehow all the source control tabs moved themselves to the search section.  I must have accidentally clicked and dragged when investigating the new gitlens changes.

**Comment by mkosir** [comment]


**Reply by justinrobots** [reply]
Hey @mkosir, it looks like you have your primary `Source Control` view hidden in the side bar. That is the view that has the menu for revealing GitLens and other SCM views, so you'll need to unhide it from the side bar first.

<img width="462" alt="image" src="https://github.com/user-attachments/assets/4ecc0fbc-f57d-48c2-a824-ea299e8576d4">

**Comment by MrSuddenJoy** [comment]
Just updated, tested **both** views and here is what I have to say:

Due to the nature of how I work, old way of displaying things suits me better. However Im able to see the strenghts of the new layout (if it may be called that).
